api/types/src/transaction.rs

```diff
@@ -109,6 +109,33 @@ impl Transaction {
             Transaction::GenesisTransaction(_) => 0,
         }
     }
+
+    pub fn success(&self) -> bool {
+        match self {
+            Transaction::UserTransaction(txn) => txn.info.success,
+            Transaction::BlockMetadataTransaction(txn) => txn.info.success,
+            Transaction::PendingTransaction(_txn) => false,
+            Transaction::GenesisTransaction(txn) => txn.info.success,
+        }
+    }
+
+    pub fn is_pending(&self) -> bool {
```

Comment:
nit: you could probably clean this up a little and return something like `matches!(self, Transaction::PendingTransaction(_)`)?